wire-ios/Wire-iOS/Sources/AnalyticsTrackingManager/TrackingManager.swift

```diff
@@ -19,35 +19,65 @@
 import avs
 import Foundation
 import WireCommonComponents
+import WireDomain
 import WireLogging
 import WireSyncEngine
 
 final class TrackingManager: TrackingInterface {
 
+    private let sharedUserDefaults: UserDefaults
     private let sessionManager: SessionManager
     private var observerToken: NSObjectProtocol?
 
-    init(sessionManager: SessionManager) {
+    private var journal: Journal? {
+        sessionManager.accountManager.selectedAccount.map { selectedAccount in
+            Journal(
+                userID: selectedAccount.userIdentifier,
+                storage: sharedUserDefaults
+            )
+        }
+    }
+
+    init(
+        sharedUserDefaults: UserDefaults,
+        sessionManager: SessionManager
+    ) {
+        self.sharedUserDefaults = sharedUserDefaults
         self.sessionManager = sessionManager
 
-        AVSFlowManager.getInstance()?.setEnableMetrics(!isAnalyticsDisabled)
+        migrateFromLegacyStorageIfNeeded()
+
         self.observerToken = NotificationCenter.default.addObserver(
             forName: FlowManager.AVSFlowManagerCreatedNotification,
             object: nil,
             queue: .main,
             using: { [weak self] _ in
                 guard let self else { return }
-                AVSFlowManager.getInstance()?.setEnableMetrics(!isAnalyticsDisabled)
+                AVSFlowManager.getInstance()?.setEnableMetrics(isAnalyticsTrackingEnabled)
             }
         )
+
+        AVSFlowManager.getInstance()?.setEnableMetrics(isAnalyticsTrackingEnabled)
     }
 
-    var doesUserConsentPreferenceExist: Bool {
-        ExtensionSettings.shared.disableAnalyticsSharing != nil
+    private var doesUserConsentPreferenceExist: Bool {
```

Comment:
todo: prefer a method, when property has a side effect.

Reply:
removed the migration/side effect